xbmc/pvr/windows/GUIViewStatePVR.cpp

```diff
@@ -77,8 +77,10 @@ void CGUIViewStateWindowPVRGuide::SaveViewState(void)
 
 CGUIViewStateWindowPVRTimers::CGUIViewStateWindowPVRTimers(const int windowId, const CFileItemList& items) : CGUIViewStatePVR(windowId, items)
 {
-  AddSortMethod(SortByLabel, 551, LABEL_MASKS("%L", "%I", "%L", ""));   // FileName, Size | Foldername, empty
-  AddSortMethod(SortByDate, 552, LABEL_MASKS("%L", "%J", "%L", "%J"));  // FileName, Date | Foldername, Date
+  SortAttribute sortAttributes(CSettings::Get().GetBool("filelists.ignorethewhensorting") ? SortAttributeIgnoreArticle : SortAttributeNone);
+  sortAttributes = (SortAttribute)((int)sortAttributes | SortAttributeIgnoreFolders);
```

Comment:
Please use a c++ cast instead of the c cast. Not <code>(int)sortAttributes</code>, but <code>static_cast\<int\>(sortAttributes)</code>.